### ISSUE TITLE
Panic on startup

### ISSUE THREAD

**Issue by meling** [stefanwichmann/kelvin#12]
Seems that `light.TargetLightState` is not being initialized with any color values prior to calling `light.setLightState(light.TargetLightState)`. Tried to understand what is the intention, but couldn't quite see it with minimal effort.

Could I suggest adding some unit tests, and running a linter (https://github.com/dominikh/go-tools/tree/master/cmd/megacheck or golint). 

```
INFO[2017/03/08 21:11:32] 💡 Light Hue color spot Hallway 2 spot 3 is not associated to any schedule. Ignoring...
INFO[2017/03/08 21:11:32] 💡 Light Hue beyond down 8 is not associated to any schedule. Ignoring...
INFO[2017/03/08 21:11:32] 💡 Light Hue color Hallway 1 spot 3 is not associated to any schedule. Ignoring...
INFO[2017/03/08 21:11:32] 💡 Light Hue color spot Hallway 2 spot 2 is not associated to any schedule. Ignoring...
INFO[2017/03/08 21:11:32] 💡 Light Hue color spot Kid's Bath 3 is not associated to any schedule. Ignoring...
INFO[2017/03/08 21:11:32] 💡 Light Hue beyond up 4 is not associated to any schedule. Ignoring...
INFO[2017/03/08 21:11:32] 💡 Light Hue beyond down 1 just appeared. Initializing state to 0K at 0% brightness.
INFO[2017/03/08 21:11:32] 💡 Light Hue color spot Hallway 2 spot 2 is not associated to any schedule. Ignoring...
INFO[2017/03/08 21:11:32] 💡 Light Hue color spot Kid's Bath 2 is not associated to any schedule. Ignoring...
panic: runtime error: index out of range

goroutine 390 [running]:
main.(*Light).setLightState(0xc42061a000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/meling/Dropbox/work/go/src/github.com/stefanwichmann/kelvin/light.go:209 +0x9e0
main.(*Light).update(0xc42061a000, 0x2, 0x0)
	/Users/meling/Dropbox/work/go/src/github.com/stefanwichmann/kelvin/light.go:149 +0x1de
main.(*Light).updateCyclic(0xc42061a000, 0xc42013a8c0)
	/Users/meling/Dropbox/work/go/src/github.com/stefanwichmann/kelvin/light.go:88 +0x2b8
created by main.main
	/Users/meling/Dropbox/work/go/src/github.com/stefanwichmann/kelvin/kelvin.go:82 +0x590
```

**Comment by stefanwichmann** [comment]
Thanks for reporting this! I'll look into the problem tonight.

It would be really helpful for debugging, if you could run kelvin again with the `-debug` parameter. This should give me a little more context.

Apart from this I will give the mentioned linter a try. `go vet` is already part of the deployment pipeline, runs with every build (see .travis.yml) and any detected correctness problem will result in a build breaker. The main purpose of `golint` is to report styling errors and probably won't help here.

Unit tests are a tricky beast if you are dealing with an external, changing api the produces different results for different light bulbs based an individual, user configured schedules... Nevertheless you are right, they might by really helpful. :-)

**Comment by meling** [comment]
Thanks for taking a look. I'm out of the house for a while. Should be home again in a few days.

**Comment by stefanwichmann** [comment]
Ok, I looked into the issue and am suspecting a (hopefully) rare timing issue. This has been fixed.

To be really sure this was your problem I would ask you to try kelvin again when you come home and attach the debug logs. Just to be sure. As soon as we can confirm the fix, I will push out a new release.

Thanks for your patience!

**Comment by meling** [comment]
I did a quick test again now with the `-debug` flag. It no longer throws a panic, but it also never stops emitting any output. It looks like an infinite loop as the output keeps repeating the same type of output. I have a large sample of a log file, but not sure it makes sense to paste it in here. Instead I've shared the log file through dropbox https://www.dropbox.com/s/9acjgkkavx1is5z/kelvin.log?dl=0. I may delete the file in a week or so, so you may want to download it for your analysis ;-) Thanks for taking the time to look into this. I'll be happy to help with further debug efforts as my time permits. (I'll be away from my house for a year starting August 17, so I won't have access to my home's Hue lighting.)

**Comment by stefanwichmann** [comment]
Wow, that's a massive log file! I saved a local copy so you can reclaim that precious dropbox space.

The excessive logging of ignored lights was another issue I fixed. I control all my lights with kelvin, so this one slipped through.

As mentioned before I released a new version (v1.0.3) containing these fixes. If you find the time before leaving for a year, give it a try! If not, see you in one year. You will find a vastly improved kelvin bot by then. ;-)

**Comment by meling** [comment]
Thanks for the fixes. I tried master branch just now, and no panics anymore. I didn't test the different features extensively except to see that it recognizes when I change the light using a switch. I won't have time to do more testing at this time, but it looks like the issue has been fixed and so feel free to close this one. See you next year ;-)
